### DISCUSSION TITLE
err apt

### DISCUSSION THREAD

**Discussion by PNM-nasa** [termux/termux-packages#17421] [Q&A]
I can't use command
`apt upgrade`
`apt install`
`E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)`
I read and tried https://github.com/termux/termux-packages/issues/5783 but it didn't work
And i used chat gpt(bing)

The error message Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1) indicates that the dpkg command encountered an error while processing a package. This error can occur for a variety of reasons, such as issues with package dependencies, incorrect file permissions, or problems with the package itself.

Here are a few suggestions that might help you resolve this issue:

Try running the apt-get update and apt-get upgrade commands to update your package lists and upgrade any outdated packages on your system.

If the issue persists, you can try using the dpkg --configure -a command to attempt to fix any broken packages.

If the above steps don’t work, you can try manually removing the problematic package using the dpkg --remove --force-remove-reinstreq package_name command, replacing package_name with the name of the package that’s causing the issue. After removing the package, you can try reinstalling it using the apt-get install package_name command.

I hope these suggestions help you resolve the issue. Let me know if you have any further questions or if there’s anything else I can help with.
(my english isn't very good)

**Comment by sylirre** [comment]
Without full error message it is not possible to say what went wrong and how to fix it.

`E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)` - is not a complete error message. There many reasons why `dpkg` can return error code `1`.

**Reply by PNM-nasa** [reply]
~ $ apt install python2 -y
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python2 is already the newest version (2.7.18-13).
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
2 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up liblua51-static (5.1.5) ...
update-alternatives: error: alternative path /data/data/com.termux/files/usr/bin/lua5.1 doesn't exist
update-alternatives: error: alternative path /data/data/com.termux/files/usr/bin/luac5.1 doesn't exist
dpkg: error processing package liblua51-static (--configure):
 installed liblua51-static package post-installation script subprocess returned error exit status 2
Setting up liblua52-static (5.2.4-8) ...                                         update-alternatives: error: alternative path /data/data/com.termux/files/usr/bin/lua5.2 doesn't exist
update-alternatives: error: alternative path /data/data/com.termux/files/usr/bin/luac5.2 doesn't exist
dpkg: error processing package liblua52-static (--configure):
 installed liblua52-static package post-installation script subprocess returned error exit status 2
Errors were encountered while processing:                                         liblua51-static
 liblua52-static
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)

**Comment by bloodykheeng** [comment]
Just install curl

```
apt-get install curl

```
